### PR TITLE
Chat structure started

### PR DESCRIPTION
**Pull request by ObiTracks**
Not gonna lie it was pretty difficult even getting the basics set up despite the tutorial - finding all the right imports and classes so there isn't much now except a test to query anany chat.

### CODE REVIEW

**Comment by JoseGrangerSalgado** [comment]
LGTM As the basics.

**Review by JoseGrangerSalgado — APPROVED** [review]

File: src/main/java/com/google/sps/data/Chat.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main.java.com.google.sps.data;
+
+/** An item on a todo list. */
+public final class Chat {
+
+  private final long id;
```
Comment (josueetcom) [inline src/main/java/com/google/sps/data/Chat.java]: Probably want a `String[]` of `participants` too.

File: src/main/java/com/google/sps/servlets/ChatServlet.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+@WebServlet("/chats")
+public class ChatServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // OBJECTIVE: Create a chat between users
+    // Step 1: Get the user and the partner ids
+    String participant1 = Jsoup.clean(request.getParameter("userid"), Whitelist.none());
+    String participant2 = Jsoup.clean(request.getParameter("userid"), Whitelist.none());
```
Comment (josueetcom) [inline src/main/java/com/google/sps/servlets/ChatServlet.java]: Hmm, not the way I'd recommend getting a specific chat (which would use the chat `id`) or a list of chats for a specific user (which uses just a single `userId`)

File: src/main/java/com/google/sps/servlets/ChatServlet.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    // Step 3: Create a new chat if it doesn't exist
+        // String chatid = participant1 + "-" + participant2;
+        // String[] participants = {participant1, participant2};
+        // ArrayList<String> messages = new ArrayList<String>();
+    // Query<Entity> query = Query.newEntityQueryBuilder();
+    // QueryResults<Entity> results = datastore.run(query);
+    // get form information
+
+    response.sendRedirect("dashboard.html");
```
Comment (josueetcom) [inline src/main/java/com/google/sps/servlets/ChatServlet.java]: `GET` requests shouldn't be causing a redirect, they should be returning a resource.

**Review by josueetcom — CHANGES_REQUESTED** [review]

**Review by jpaints3 — APPROVED** [review]
Yeah, I agree with Josue's comments and I think that a query is a right way to go. I think you are on the right track and I'm going to approve for the sake of time. I think if you get the other changes in, it will be a good or for the committed goal of creating a chat structure.

**Review by josueetcom — APPROVED** [review]
Approving this to get you unblocked but please address any requested changes before merging.

**Pull request merged**